core/src/main/java/org/apache/accumulo/core/util/LocalityGroupUtil.java

```diff
@@ -377,11 +377,11 @@ public static void seek(FileSKVIterator reader, Range range, String lgName,
     if (lgName == null) {
       // this is the default locality group, create a set of all families not in the default group
       Set<ByteSequence> nonDefaultFamilies = new HashSet<>();
-      for (Entry<String,ArrayList<ByteSequence>> entry : localityGroupCF.entrySet()) {
-        if (entry.getKey() != null) {
-          nonDefaultFamilies.addAll(entry.getValue());
+      localityGroupCF.forEach((k, v) -> {
+        if (k != null) {
+          nonDefaultFamilies.addAll(v);
         }
-      }
+      });
```

Comment:
Is it possible to have null keys in the first place? If not, that check is probably redundant, and we could just stream the values. Something like `localityGroupCF.values().stream().forEach(nonDefaultFamilies::addAll);`

If it is possible to have null keys, then your current code is probably fine.

Reply:
Looking at the code that calls this, it looks like the key can be null. 
https://github.com/apache/accumulo/blob/0781550076f04d12716650fd64881d8b9d041afa/core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java#L260

It pulls `lgname` from the Map and then passes it in with the map. Based on the check before the loop, `if (lgName == null)` there will be at least one key that is definitely null.

Reply:
Thanks for the analysis, @milleruntime . HashMap does support a single null key and multiple null values, but that doesn't necessarily mean it's a good idea. However, if there's any improvements needed to avoid a null key, it's unrelated to this PR.